breaking changes in inquirer 12.11.0

In 12.10.0 the prefilled answers to `prompt` can be `Partial<C>` but with 12.11.0 TypeScript reports error:

```
 src/bin/execute-configs.ts(105,3): error TS2322: Type '{ displayName?: string | undefined; technology?: string | undefined; iconUrl?: string | undefined; }' is not assignable to type 'WidgetConfig'.
   Types of property 'displayName' are incompatible.
     Type 'string | undefined' is not assignable to type 'string'.
       Type 'undefined' is not assignable to type 'string'.
 src/bin/execute-configs.ts(133,3): error TS2322: Type '{ managementApiEndpoint?: string | undefined; resourceId?: string | undefined; apiVersion?: string | undefined; }' is not assignable to type 'ServiceInformation'.
   Types of property 'managementApiEndpoint' are incompatible.
     Type 'string | undefined' is not assignable to type 'string'.
       Type 'undefined' is not assignable to type 'string'.
```

Our current code:

``` typescript
export const promptWidgetConfig = async (partial: Partial<WidgetConfig>): Promise<WidgetConfig> => {
  const inquirerImport = await import("inquirer");
  const inquirer = inquirerImport.default;
  return inquirer.prompt(
    [
      {
        name: "displayName",
        type: "input",
        message: fieldIdToName.displayName,
        validate: validateWidgetConfig.displayName,
      },
      {
        name: "technology",
        type: "list",
        message: fieldIdToName.technology,
        choices: [
          { name: "React", value: "react" },
          { name: "Vue", value: "vue" },
          { name: "TypeScript", value: "typescript" },
        ],
      },
    ],
    partial,
  );
};
```

--- a/packages/inquirer/inquirer.test.ts
+++ b/packages/inquirer/inquirer.test.ts
@@ -846,6 +846,120 @@ describe('inquirer.prompt(...)', () => {
       expectTypeOf(answers.prefilled).toEqualTypeOf<{ nested: string }>();
       expectTypeOf(answers.prefilled).not.toBeAny();
     });
+
+    describe('Partial<T> prefilled answers (issue #1888)', () => {
+      it('should accept Partial-like objects and strip undefined from types', async () => {
+        // Use a partial object without explicit Partial<T> type annotation
+        const partial = {
+          displayName: 'Custom Widget',
+          technology: 'vue',
+        };
+
+        const result = await inquirer.prompt(
+          [
+            {
+              name: 'displayName',
+              type: 'stub',
+              message: 'Widget Display Name',
+              answer: 'My Widget',
+            },
+            {
+              name: 'technology',
+              type: 'stub',
+              message: 'Technology',
+              answer: 'react',
+            },
+            {
+              name: 'iconUrl',
+              type: 'stub',
+              message: 'Icon URL',
+              answer: 'https://example.com/icon.png',
+            },
+          ],
+          partial,
+        );
+
+        // Runtime behavior: prefilled values are used
+        expect(result.displayName).toEqual('Custom Widget');
+        expect(result.technology).toEqual('vue');
+        expect(result.iconUrl).toEqual('https://example.com/icon.png');
+
+        // Type checks: result properties are string, not string | undefined
+        expectTypeOf(result.displayName).toEqualTypeOf<string>();
+        expectTypeOf(result.technology).toEqualTypeOf<string>();
+        expectTypeOf(result).not.toBeAny();
+      });
+
+      it('should handle prefilled answers with some fields undefined', async () => {
+        const partial = {
+          field1: 'value1',
+          field2: undefined as string | undefined,
+        };
+
+        const result = await inquirer.prompt(
+          [
+            {
+              name: 'field1',
+              type: 'stub',
+              message: 'Field 1',
+              answer: 'default1',
+            },
+            {
+              name: 'field2',
+              type: 'stub',
+              message: 'Field 2',
+              answer: 'default2',
+            },
+          ],
+          partial,
+        );
+
+        // field1 should be prefilled, field2 should be prompted
+        expect(result.field1).toEqual('value1');
+        expect(result.field2).toEqual('default2');
+        expectTypeOf(result.field1).toEqualTypeOf<string>();
+        expectTypeOf(result.field2).toEqualTypeOf<string>();
+        expectTypeOf(result).not.toBeAny();
+      });
+
+      it('should handle multiple prefilled fields of same type', async () => {
+        const partial = {
+          name: 'Custom Name',
+          description: 'Custom Description',
+        };
+
+        const result = await inquirer.prompt(
+          [
+            {
+              name: 'name',
+              type: 'stub',
+              message: 'Name',
+              answer: 'Default Name',
+            },
+            {
+              name: 'description',
+              type: 'stub',
+              message: 'Description',
+              answer: 'Default Description',
+            },
+            {
+              name: 'email',
+              type: 'stub',
+              message: 'Email',
+              answer: 'default@example.com',
+            },
+          ],
+          partial,
+        );
+
+        expect(result.name).toEqual('Custom Name');
+        expect(result.description).toEqual('Custom Description');
+        expect(result.email).toEqual('default@example.com');
+        expectTypeOf(result.name).toEqualTypeOf<string>();
+        expectTypeOf(result.description).toEqualTypeOf<string>();
+        expectTypeOf(result).not.toBeAny();
+      });
+    });
   });
 
   describe('#registerPrompt()', () => {

--- a/packages/inquirer/src/index.ts
+++ b/packages/inquirer/src/index.ts
@@ -124,6 +124,10 @@ export function createPromptModule<
   function promptModule<A extends Answers>(
     questions: PromptSession<A>,
     answers?: Partial<A>,
+  ): PromptReturnType<A>;
+  function promptModule<A extends Answers>(
+    questions: PromptSession<A>,
+    answers?: Partial<A>,
   ): PromptReturnType<A> {
     const runner = new PromptsRunner<A>(promptModule.prompts, opt);
 

--- a/packages/inquirer/src/types.ts
+++ b/packages/inquirer/src/types.ts
@@ -69,7 +69,13 @@ type WidenAnswerLiterals<T> = T extends string
             : T extends Array<infer U>
               ? Array<WidenAnswerLiterals<U>>
               : T extends Record<string, unknown>
-                ? { [K in keyof Mutable<T>]: WidenAnswerLiterals<Mutable<T>[K]> }
+                ? {
+                    [K in keyof Mutable<T>]: Mutable<T>[K] extends infer V
+                      ? V extends undefined
+                        ? never
+                        : WidenAnswerLiterals<V>
+                      : never;
+                  }
                 : T;
 
 type MergeAnswerObjects<Base, Override> = Prettify<Omit<Base, keyof Override> & Override>;

